src/Modules/Workspaces/Commands.Workspaces/GetPowerBIWorkspace.cs

```diff
@@ -125,5 +147,36 @@ public override void ExecuteCmdlet()
                 this.Logger.WriteObject(workspaces, true);
             }
         }
+
+        private void ExecuteCmdletWithAll()
+        {
+            var allWorkspaces = new List<Workspace>();
+            using (var client = this.CreateClient())
+            {
+                var top = 5000;
+                var skip = 0;
+                var workspacesCountBeforeApiCall = 0L;
+                var workspacesCountAfterApiCall = 0L;
+
+                // GetWorkspacesAsAdmin is called with 5000 as the top which retrieves 5000 workspaces in each call.
+                // Loop will terminate when the workspaces retrieved are less than 5000.
+                do
+                {
+                    var result = client.Workspaces.GetWorkspacesAsAdmin(expand: "users", filter: this.Filter, top: top, skip: skip);
+                    allWorkspaces.AddRange(result);
+                    workspacesCountBeforeApiCall = workspacesCountAfterApiCall;
+                    workspacesCountAfterApiCall = allWorkspaces.Count;
+                    skip += top;
+                } while ((workspacesCountAfterApiCall - workspacesCountBeforeApiCall) == top);
```

Comment:
Don't we just need to capture the count we got back, if it equaled top then there are more records until we get less than top we continue

Reply:
Thats right, for some reason when I was initially coding this I wasn't seeing a Count() method inside the result variable type. Probably something to do with my local project build state. I changed it now to just use the result.Count().